libs/wingsdk/src/shared-aws/api.ts

```diff
@@ -0,0 +1,40 @@
+import { IResource } from "../std";
+
+/**
+ * A shared interface for AWS api.
+ */
+export interface IAwsApi {
+  /**
+   * Get internal AWS ApiGateway
+   */
+  innerAwsApi(): any;
```

Comment:
as much as it might be tempting to simply expose these raw resources, i think that's not a very useful api because the consumer needs too much of a deep knowledge about the internal anatomy. for example, awscdk and cdktf would have a totally different internal api.


i am wondering if a better way is for each resource expose some explicit attributes. for example, for the API resource on AWS, I would expose things like `restApiId`, `stageName`, `deploymentId`, etc.

what do you think?

Reply:
I understand. 
Personally, exposing the raw resource makes it easier to work with L2 constructs from aws-cdk, but it makes the library less understandable for those without experience with CdkTF and aws-cdk.

I will make changes to expose the ids and other relevant information. 
In the case of CdkTF and aws-cdk, the naming is not always the same, so I will choose the one that I find more readable. 

For example, between `functionArn` and `arn`, I will opt for `arn` since the resource type will already be defined.
So, to obtain the function arn the code could be like this (for both cdktf and aws-cdk).

```ts
bring cloud;
bring aws;

let bucket = new cloud.Bucket();
if let b = aws.Bucket.from(bucket) {
    let functionArn = b.arn();
}
```

Reply:
i actually think that `function.arn` is nicer than `function.functionArn`, but please make a revision with some naming suggestion and i'll review.